### PR TITLE
Add tests for rewrite rules

### PR DESCRIPTION
**Pull request by adpi2**
Add tests for the documented rewrite rules.

The `rewrites / test` task will perform these steps:
  - copy the source files from `rewrites/input/`  to `rewrites/target/src-managed/`
  - compile the files in `rewrites/target/src-managed/` with `-source:3.0-migration -rewrite`
  - compare the files in `rewrites/target/src-managed/` with the files in `rewrites/check/`

It is also possible to change the migration version to 3.1 to test the `-source:3.1-migration` rules.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.metals/
+target
```
Comment (sjrd) [inline .gitignore:2]:
```suggestion
target/
```

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: Continuous Integration
+on:
+ pull_request:
```
Comment (sjrd) [inline .github/workflows/continuous-integration.yml:3]: This should probably have `branches: [master]` as well.

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+val outputDir = settingKey[File]("Directory in which the source files are rewritten")
+val checkDir = settingKey[File]("Directory that contains the expected rewritten files")
+val migration = settingKey[String]("Target Scala version - 3.0 or 3.1")
+
+val rewrites = (project in file("rewrites"))
+  .settings(
+    scalaVersion := dottyVersion,
+    migration := "3.0",
+    scalacOptions ++= Seq(s"-source:${migration.value}-migration","-rewrite"),
+    inputDir := new File(baseDirectory.value, s"src/input/scala-${migration.value}"),
```
Comment (sjrd) [inline build.sbt:15]: Consider using sbt's syntax for this:
```suggestion
    inputDir := baseDirectory.value / s"src/input/scala-${migration.value}",
```
Same below.

File: rewrites/src/check/scala-3.0/1-keywords.scala [hunk 4]
```diff
@@ -0,0 +1,5 @@
+object `given` {
```
[the remark this answers is not in the document]
Comment (adpi2) [reply rewrites/src/check/scala-3.0/1-keywords.scala:1]: I have given a number to each rewrite rule in the documentation. There is no logic behind it but it makes it easier to refer to a rule in particular. When a test fails it is easier to find the documentation that may need to be updated along with the test.

So I would prefer to keep it that way. If it proves to be hard to maintain then we would reconsider.

**Pull request merged**